projects/RabbitMQ.Client/client/impl/ModelBase.cs

```diff
@@ -332,38 +334,64 @@ private void HandleCommand(in IncomingCommand cmd)
             }
         }
 
-        protected T ModelRpc<T>(MethodBase method) where T : MethodBase
+        protected void ModelRpc<TMethod>(in TMethod method, ProtocolCommandId returnCommandId)
```

Comment:
What's the reason for removing the ProtocolCommandId from the method classes and adding it as an extra parameter?

Reply:
There are a lot of methods which are not interested in parsing the returned method (or is empty) so by passing the id, we can do the check if it is the correct response. If you want to parse the result, there are extra methods for it.